pkgs/development/python-modules/python3-application/default.nix

```diff
@@ -0,0 +1,47 @@
+{ lib, isPy3k, buildPythonPackage, fetchFromGitHub, fetchpatch, zope_interface, twisted }:
+
+buildPythonPackage rec {
+  pname = "python3-application";
+  version = "3.0.3";
+
+  disabled = !isPy3k;
+
+  src = fetchFromGitHub {
+    owner = "AGProjects";
+    repo = pname;
+    rev = version;
+    sha256 = "sha256-oscUI/Ag/UXmAi/LN1pPTdyqQe9aAfeQzhKFxaTmW3A=";
+  };
+
+  patches = [
+    # Apply bugfix commit that is not yet part of a release
+    (fetchpatch {
+      name = "fix-time-import.patch";
+      url = "https://github.com/AGProjects/${pname}/commit/695f7d769e69c84e065872ffb403157d0af282fd.patch";
```

Comment:
This should not use pname here but the string literal. If pname is ever changed the FOD is not validated and potentially we are using cached files which are not reproducible. Also I cannot click on that link to inspect the patch in the source file.

Reply:
Both are good points. I opened https://github.com/NixOS/nixpkgs/pull/139567 to change.